editor/effects.js

```diff
@@ -132,19 +132,42 @@ export default {
 		const { dispatch, getState } = store;
 		const { postId } = action;
 		const Model = wp.api.getPostTypeModel( getCurrentPostType( getState() ) );
-		new Model( { id: postId } ).destroy().done( () => {
-			dispatch( {
-				...action,
-				type: 'TRASH_POST_SUCCESS',
-			} );
-		} );
+		new Model( { id: postId } ).destroy().then(
+			() => {
+				dispatch( {
+					...action,
+					type: 'TRASH_POST_SUCCESS',
+				} );
+			},
+			( err ) => {
```

Comment:
I think I was originally going to use it for checking the dirty state, but I went a different direction. I've split this out into #1857.